packages/smooth_ui_library/lib/widgets/smooth_toggle.dart

```diff
@@ -207,7 +206,9 @@ class _SmoothToggleState extends State<SmoothToggle>
         turnState = !turnState;
       }
       turnState ? animationController.forward() : animationController.reverse();
-      widget.onChanged(turnState);
+      if (widget.onChanged != null) {
```

Comment:
I still cannot see the purpose of this widget if `onChanged` can be null.
My suggestion: make `onChanged` not nullable at all and `required`. The .0001% cases where developers don't need `onChanged`, they will have to add something like `onChanged: (bool flag) {}` to their code, which is affordable.
`onChanged` gives the widget a general purpose.